### DISCUSSION TITLE
Cannot access parent interface properties directly

### DISCUSSION THREAD

**Discussion by CREAsTIVE** [dotnet/roslyn#71529] [Q&A] [answered]
I think an example will be enough

```cs
public interface ITest
{
	protected string _name { get; set; }
	public string Name { get => _name; set => _name = value; }
}

public class Test : ITest
{
	public string _name { get; set; } = "e";

	public Test()
	{
		Console.WriteLine(((ITest)this).Name); // Works!
		Console.WriteLine(Name); // Compilation error???
		// Symbol "Name" doesnt exist in current context
	}
}

```

**Comment by huoyaoyuan** [comment]
This is a C# language question.

Interfaces implemented by a class isn't "parent" like base class.

**Reply by CREAsTIVE** [reply, marked as answer]
Oh sorry, wrong repository... Should i go to [https://github.com/dotnet/csharplang/discussions](https://github.com/dotnet/csharplang/discussions)?

**Reply by CyrusNajmabadi** [reply]
You could move it there.  But the answer would be the same as what @huoyaoyuan said.  Interface members are not part of the inheritance chain of a class (never have been, and we don't have any intent on making them so in the future).  If you want to access an interface member, you do so through a reference to that interface type (or a derived interface), not an a reference to an implementing class or struct.

Thanks!
